storage/src/adb/immutable.rs

```diff
@@ -302,14 +311,41 @@ impl<E: RStorage + Clock + Metrics, K: Array, V: Codec, H: CHasher, T: Translato
         assert_eq!(log_size, leaf_pos_to_num(mmr.size()).unwrap());
         assert_eq!(log_size, locations.size().await?);
 
-        Ok(log_size)
+        Ok((log_size, oldest_retained_loc.unwrap_or(0)))
```

Comment:
Does it make sense to `prune` mmr/locations based on the oldest_retained we have here?

Reply:
You mean to repair any failures that occurred during a prune?  We could, but it would be corrected anyway next time prune() is called.

Reply:
Yes, repair failures.

I see...we gracefully handle things before the oldest because we need to support delayed cleanup of the `index` (rather than asserting something in index must be fetchable).

This should be fine...I think it would be worth adding a comment before this line commenting on this case.

Reply:
Jk, just saw you wrote about in prune. Disregard.